[WIP] add Block THT transformers

lorem-ipsum opened the pull request:
This is the first revision of my attempt to script the collection of THT transformers from Block. Currently it contains the packages used in the FL series. The website is not really good, the best overview for the dimensions is the [complete catalogue](https://www.block.eu/fileadmin/Blaetterkatalog_und_ebook/eBook.pdf) (~60MB), the corresponding dimensions are on page 281. My corresponding python mess can be found at https://github.com/lorem-ipsum/kicad-footprint-generator/tree/block (no PR yet, will post it once the path is clear).

At the current state I have two questions:
1. Any comments about the footprint in general? Currently it is very simple, containing only the required markers, but no extra drawings. Some of the currently existing footprints have a lot of extra bells and whistles (e.g. marking the coils on fab layer, writing Pri and Sec on the Silk, etc.). Should I add something like this?
2. Probably more important: How to name them correctly? The current naming uses the core type as they appeared to have a unique housing each. However I already found the obviously existing exception from the rule as the FL 52 and the FLD 48 (on page 285) share the same core, but have slightly different dimensions. The obvious alternative is to use the Type name (e.g. FL 52). This is definitely unique, but will create many duplicate footprints when adding all transformers (which I intend to do).

Example screenshot of one of the footprints:
![Screenshot_20190801_225902](https://user-images.githubusercontent.com/1010629/62327099-fdb0a700-b4af-11e9-8530-9fc22b4e846b.png)

---

All contributions to the kicad library must follow the [KiCad library convention](http://kicad-pcb.org/libraries/klc/)

Thanks for creating a pull request to contribute to the KiCad libraries! To speed up integration of your PR, please check the following items:

- [ ] Provide a URL to a datasheet for the footprint(s) you are contributing
- [ ] An example screenshot image is very helpful 
- [ ] If there are matching symbol or 3D model pull requests, provide link(s) as appropriate
- [ ] Check the output of the Travis automated check scripts - fix any errors as required
- [ ] Give a reason behind any intentional library convention rule violation.

evanshultz commented:
1. I prefer simplicity. No need to add this extra stuff. As you can see some interpretations vary, but KLC does not require this (in fact it goes against extraneous markings). IMO I would personally remove this stuff from all existing footprints unless somebody objects.
1. Since the 3D model is unique for each, I don't see any duplicated dimensions (LxWxH). Since footprints are relatively small files, if having all of them in the library is useful to KiCad users then I would include each one. I would also think basic dimensions would be nice since a generic symbol could fit many footprints and so the user will be able to get a sense of the size (and VA) if that's in the footprint name. Something like `Transformer_BLOCK_FL2_53x44x17.6mm`. https://github.com/KiCad/kicad-footprints/pull/1602, and the issues I linked, contain some discussion about transformer footprint naming. As these are off-the-shelf and not custom transformers, though, not all the rules apply.

When you add the URL, please include the page number of the PDF so that a navigating a big file like the one you linked isn't tough. Append `#page=281` to the URL, for the FL series.

In addition, two other comemnts:
1. I don't see fab or silk pin 1 marks.
1. Typically the value is below the part and not inside.

lorem-ipsum commented:
I now changed the naming scheme to `Transformer_Block_{series}-{power}[-outputs]_{L}x{W}x{H}mm` and created all series which share the dimension labeling with the FL and the VB series. I also added pin 1 makers on fab and silk and moved the value below the package. I had it inside as the components are rather large, so there is enough of space.

I also got the unique datasheet links for each package using a small script. I always used the 12V variant as this exists for all transformers (output voltage has no effect on the package).

Below a reference screenshot of one of the smallest transformers:
![Screenshot_20190808_132825](https://user-images.githubusercontent.com/1010629/62699924-e0646700-b9e0-11e9-8474-e1f7b272e5bd.png)

lorem-ipsum commented:
another question which I stumbled upon when reading this issue: https://github.com/KiCad/kicad-symbols/issues/146. How should the pins be numbered? The datasheet gives no numbering scheme. Currently they are just numbered counter clockwise. The suggestion in this issue is to separate the numbers between primary and secondary side. Should I do this?
